### PR TITLE
✨ NEW: Add `verdi database summary`

### PR DESCRIPTION
**Pull request by chrisjsewell**
Prints a summary of the count of each entity and, with `-v`, additional summary of the unique identifiers for some entities.

Example:

```console
$ verdi database summary -v
Users:
  count: 1
  emails:
  - tests@aiida.mail
Computers:
  count: 1
  names:
  - localhost-test
Nodes:
  count: 4
  node_types:
  - data.code.Code.
  - data.dict.Dict.
  - process.calculation.calcjob.CalcJobNode.
  process_types:
  - aiida.calculations:templatereplacer
Groups:
  count: 4
  type_strings:
  - core
Comments:
  count: 10
Logs:
  count: 30
```

### CODE REVIEW

**Comment by chrisjsewell** [comment]
weird this works locally:

```
 run_cli_command = <function run_cli_command.<locals>._run_cli_command at 0x7fb94db06b00>

    @pytest.mark.usefixtures('aiida_profile')
    def tests_database_summary(run_cli_command):
        from aiida import orm
>       node = orm.Dict().store()

tests/cmdline/commands/test_database.py:188: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
aiida/orm/nodes/data/dict.py:60: in __init__
    super().__init__(**kwargs)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <[AttributeError("'Dict' object has no attribute '_backend_entity'") raised in repr()] Dict object at 0x7fb94d860f90>
backend = <aiida.orm.implementation.sqlalchemy.backend.SqlaBackend object at 0x7fb94d860c90>
user = None, computer = None, kwargs = {}

    def __init__(self, backend=None, user=None, computer=None, **kwargs):
        backend = backend or get_manager().get_backend()
    
        if computer and not computer.is_stored:
            raise ValueError('the computer is not stored')
    
        computer = computer.backend_entity if computer else None
        user = user.backend_entity if user else User.objects(backend).get_default()
    
        if user is None:
>           raise ValueError('the user cannot be None')
E           ValueError: the user cannot be None
```

I guess the default pgtest temporary database is created with a user, but the

File: aiida/cmdline/commands/cmd_database.py [hunk 1]
```diff
@@ -194,3 +194,51 @@ def detect_invalid_nodes():
...
+    """Summarise the entities in the database."""
+    from aiida.orm import QueryBuilder, Node, Group, Computer, Comment, Log, User
+    data = {}
+
+    # User
+    count = QueryBuilder().append(User).count()
+    data['Users'] = {'count': count}
+    if verbose:
+        emails = QueryBuilder().append(User, project=['email']).distinct().all(flat=True)
+        data['Users']['emails'] = emails
```
Comment (yakutovicha) [inline aiida/cmdline/commands/cmd_database.py:211]:
```suggestion
    user = QueryBuilder().append(User, project=['email'])
    data['Users'] = {'count': user.count()}
    if verbose:
        data['Users']['emails'] = user.distinct().all(flat=True)
```

**Review by yakutovicha — COMMENTED** [review]
Looks good overall, I put a suggestion below on how I would simplify a bit the code. I provided only one example, but the same can be applied to computers, nodes, etc.

File: aiida/cmdline/commands/cmd_database.py [hunk 2]
```diff
@@ -194,3 +194,51 @@ def detect_invalid_nodes():
...
+
+    # Computer
+    count = QueryBuilder().append(Computer).count()
+    data['Computers'] = {'count': count}
+    if verbose:
+        names = QueryBuilder().append(Computer, project=['name']).distinct().all(flat=True)
```
Comment (chrisjsewell) [inline aiida/cmdline/commands/cmd_database.py:217]:
```suggestion
    query_comp QueryBuilder().append(Computer, project=['name'])
    data['Computers'] = {'count': query_comp.count()}
    if verbose:
        names = query_comp.distinct().all(flat=True)
```

File: aiida/cmdline/commands/cmd_database.py [hunk 3]
```diff
@@ -194,3 +194,51 @@ def detect_invalid_nodes():
...
+        node_types = QueryBuilder().append(Node, project=['node_type']).distinct().all(flat=True)
+        data['Nodes']['node_types'] = node_types
+        process_types = QueryBuilder().append(Node, project=['process_type']).distinct().all(flat=True)
+        data['Nodes']['process_types'] = [p for p in process_types if p]
+
+    # Group
+    count = QueryBuilder().append(Group).count()
+    data['Groups'] = {'count': count}
+    if verbose:
+        type_strings = QueryBuilder().append(Group, project=['type_string']).distinct().all(flat=True)
```
Comment (chrisjsewell) [inline aiida/cmdline/commands/cmd_database.py:233]:
```suggestion
    query_group = QueryBuilder().append(Group, project=['type_string'])
    data['Groups'] = {'count': query_group.count()}
    if verbose:
        type_strings = query_group.distinct().all(flat=True)
```

File: aiida/cmdline/commands/cmd_database.py [hunk 4]
```diff
@@ -194,3 +194,50 @@ def detect_invalid_nodes():
...
+    data['Users'] = {'count': query_user.count()}
+    if verbose:
+        data['Users']['emails'] = query_user.distinct().all(flat=True)
+
+    # Computer
+    query_comp = QueryBuilder().append(Computer, project=['name'])
+    data['Computers'] = {'count': query_comp.count()}
+    if verbose:
+        names = query_comp.distinct().all(flat=True)
+        data['Computers']['names'] = names
```
Comment (yakutovicha) [inline aiida/cmdline/commands/cmd_database.py:217]:
```suggestion
        data['Computers']['names'] = query_comp.distinct().all(flat=True)
```

File: aiida/cmdline/commands/cmd_database.py [hunk 5]
```diff
@@ -194,3 +194,50 @@ def detect_invalid_nodes():
...
+        data['Nodes']['node_types'] = node_types
+        process_types = QueryBuilder().append(Node, project=['process_type']).distinct().all(flat=True)
+        data['Nodes']['process_types'] = [p for p in process_types if p]
+
+    # Group
+    query_group = QueryBuilder().append(Group, project=['type_string'])
+    data['Groups'] = {'count': query_group.count()}
+    if verbose:
+        type_strings = query_group.distinct().all(flat=True)
+        data['Groups']['type_strings'] = type_strings
```
Comment (yakutovicha) [inline aiida/cmdline/commands/cmd_database.py:233]:
```suggestion
        data['Groups']['type_strings'] =  query_group.distinct().all(flat=True)
```

**Review by yakutovicha — CHANGES_REQUESTED** [review]
two minor details and we are good to go.

File: aiida/cmdline/commands/cmd_database.py [hunk 6]
```diff
@@ -194,3 +194,48 @@ def detect_invalid_nodes():
...
+        node_types = QueryBuilder().append(Node, project=['node_type']).distinct().all(flat=True)
+        data['Nodes']['node_types'] = node_types
+        process_types = QueryBuilder().append(Node, project=['process_type']).distinct().all(flat=True)
+        data['Nodes']['process_types'] = [p for p in process_types if p]
+
+    # Group
+    query_group = QueryBuilder().append(Group, project=['type_string'])
+    data['Groups'] = {'count': query_group.count()}
+    if verbose:
+        data['Groups']['type_strings'] =  query_group.distinct().all(flat=True)
```
Comment (chrisjsewell) [inline aiida/cmdline/commands/cmd_database.py:231]:
```suggestion
        data['Groups']['type_strings'] = query_group.distinct().all(flat=True)
```

**Review by yakutovicha — APPROVED** [review]
Looks good, merge it!

**Pull request merged**